File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/dto/waitlist/WaitlistResponse.java
package com.hcmute.reservation.dto.waitlist;

import com.hcmute.reservation.model.enums.WaitlistStatus;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class WaitlistResponse {
    private Long waitlistId;
    private String customerName;
    private String customerPhone;
    private int guestCount;
    private boolean allowShortSeating;
    private LocalDateTime joinedAt;
    private WaitlistStatus status;
    private boolean readyToSeat;
    private String seatingType;
    private LocalDateTime suggestedAvailableUntil;
    private List<Long> suggestedTableIds;
}

File: desktop-staff/reservation-winforms/DTO/waitList/WaitlistResponse.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace reservation_winforms.DTO.waitList
{
    public class WaitlistResponse
    {
        [JsonProperty("waitlistId")]
        public long WaitlistId { get; set; }

        [JsonProperty("customerName")]
        public string CustomerName { get; set; }

        [JsonProperty("customerPhone")]
        public string CustomerPhone { get; set; }

        [JsonProperty("guestCount")]
        public int GuestCount { get; set; }

        [JsonProperty("allowShortSeating")]
        public bool AllowShortSeating { get; set; }

        [JsonProperty("joinedAt")]
        public DateTime JoinedAt { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; } // Enum WaitlistStatus

        [JsonProperty("readyToSeat")]
        public bool ReadyToSeat { get; set; }

        [JsonProperty("seatingType")]
        public string SeatingType { get; set; }

        [JsonProperty("suggestedAvailableUntil")]
        public DateTime? SuggestedAvailableUntil { get; set; } // Có thể null

        [JsonProperty("suggestedTableIds")]
        public List<long> SuggestedTableIds { get; set; }
    }
}